core/templates/components/forms/schema-based-editors/schema-based-editor.directive.ts

```diff
@@ -67,7 +67,13 @@ angular.module('oppia').directive('schemaBasedEditor', [
       },
       template: require('./schema-based-editor.directive.html'),
       controllerAs: '$ctrl',
-      controller: [function() {}]
+      controller: [
+        '$rootScope', function($rootScope) {
+          let ctrl = this;
+          ctrl.$onInit = function() {
+            $rootScope.$applyAsync();
```

Comment:
how the init function is helping with change detection?

Reply:
Please refer https://github.com/oppia/oppia/pull/12602#discussion_r621910944

When the schema-based-editor is initialized, the view is not updating.
So, by calling applyAsync(), I am explicitly asking ajs to update the view after schema-based-editor is initialized.

Reply:
Hm, I'm a bit confused. Do we know why change detection is not happening here? We previously only did this in places that involved some HTTP requests (see [here](https://github.com/oppia/oppia/wiki/Angular-Migration) -- "Why do we need $rootScope.$applyAsync with HTTP requests?").

Reply:
We have used downgradeModule to bootstrap the oppia hybrid app which means we downgrade our angular components/services and use them in angularjs. Unlike upgradeModule, change of detection of ajs and angular are not linked in downgradeModule. So, we have to explicitly call change detection to make changes of in one framework reflect in other. Oppia is a large and interlinked application, so in most cases, changes done in one framework trigger change detection in the other framework as a side effect (ideally, it shouldn't), and sometimes it does not. We have to handle the other cases explicitly. In the above example, angular component is not able to detect the initialization of angularjs schema-based-directive. So, the natural question would be that we should trigger change detection in angular but we have done that in ajs instead. Yeah, but it was tricky to trigger an event after the initialization of ajs directive and listen to that event in angular component and trigger change detection in angular using change detector ref (This was my first approach). Using $rootScope.$applyAsync() here is a trick, root scope apply async is used to trigger change detection in ajs, but for some reason that I don't know, it is also triggering angular change detection (side effect). So, here I am taking advantage of this side effect of $rootScope.$applyAsync().
The credit for this fix goes to @Radesh-kumar

Reply:
> . So, we have to explicitly call change detection to make changes of in one framework reflect in other.

Why is this a special case? I don't see call to `$rootScope.$applyAsync()` in other directives for change-detection.

@srijanreddy98 Can you please check this and confirm it's okay to explicitly run digest-cycle here?

Reply:
@ashutoshc8101 Can you please add a code comment above? (As suggested by @kevintab95 [here](https://github.com/oppia/oppia/pull/12602#pullrequestreview-649705387))

Reply:
I haved added